src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (user.getLogin() == null || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new ValidationException("The value must not be a null");
+        }
+        if (!checkOnExistedLogin(user)) {
+            storageDao.add(user);
+        }
```

Comment:
This is a bug. You are checking if the user exists in the storage after you have added the user to the storage. You should check it before adding. If the user already exists, you should throw an exception.